ng2-components/ng2-activiti-form/src/services/form.service.ts

```diff
@@ -47,6 +47,23 @@ export class FormService {
                 private logService: LogService) {
     }
 
+    parseForm(json: any, data?: FormValues, readOnly: boolean = false): FormModel {
```

Comment:
This will complicate entire flow, as it will be hard to tell the parsing didn't happen. Developers will need to do additional checks to see whether result value is what they expect or not, especially if binding in Angular template. I think 'null' is perfectly valid in this case. You can always do in your code `let model = parseForm(...) || new FormModel(...)`